crates/astria-bridge-contracts/src/lib.rs

```diff
@@ -271,6 +274,13 @@ where
                     .parse()
                     .map_err(BuildError::parse_ics20_asset_source_channel)?,
             );
+
+            // USDC from Noble requires using the bech32 compat address as the sender/return
+            // address.
+            //
+            // since we're always unwrapping the asset (sending back to the originating chain), we
+            // can check that the base denom is USDC, and if it is, we know we're sending to Noble.
+            use_compat_address = ics20_asset_to_withdraw.base_denom() == "usdc";
```

Comment:
Instead of hardcoding this here, I'd prefer to add a method `fn use_compat_address(self, use_compat_address) -> Self` on the builder and let the user of this crate decide this.